.travis.yml

```diff
@@ -44,14 +44,23 @@ script:
 
   elif [ "$TEST_TYPE" = build_website ]
   then
-
+    DOCS_VERSION=${TRAVIS_BRANCH%-stable}
     cd website
     $(which npm) install
     ./setup.sh
-    if [ "$TRAVIS_PULL_REQUEST" = false ] && [ "$TRAVIS_BRANCH" = master ]; then
-      # Automatically publish the website
-      echo "machine github.com login reactjs-bot password $GITHUB_TOKEN" >~/.netrc
-      ./publish.sh
+    if [ "$TRAVIS_PULL_REQUEST" = false ]
+    then
+      if [ "$TRAVIS_BRANCH" = master ]
+      then
+        # Automatically publish the website to /next path
+        echo "machine github.com login reactjs-bot password $GITHUB_TOKEN" >~/.netrc
+        ./publish-gh-pages.sh next
```

Comment:
One other thing that sucks about having a next is that all the typo / improvement pull requests are only going to show up in ~2-4 weeks to the main website. But I guess it's the cost of doing things :p